src/compiler/checker/lines_4000_8000.rs

```diff
@@ -246,24 +324,28 @@ impl NodeBuilder {
         type_checker: &TypeChecker,
         symbol: Rc<Symbol>,
         meaning: /*SymbolFlags*/ Option<SymbolFlags>,
+        flags: Option<NodeBuilderFlags>,
         tracker: Option<&dyn SymbolTracker>,
     ) -> Option<Expression> {
-        self.with_context(tracker, |context| {
+        self.with_context(flags, tracker, |context| {
             self._symbol_to_expression(type_checker, symbol, context, meaning)
         })
     }
 
     fn with_context<TReturn, TCallback: FnOnce(&NodeBuilderContext) -> TReturn>(
         &self,
+        flags: Option<NodeBuilderFlags>,
         tracker: Option<&dyn SymbolTracker>,
         cb: TCallback,
     ) -> Option<TReturn> {
         let default_tracker: Option<DefaultNodeBuilderContextSymbolTracker> = match tracker {
             Some(_) => None,
             None => Some(DefaultNodeBuilderContextSymbolTracker::new()),
         };
-        let context =
-            NodeBuilderContext::new(tracker.unwrap_or_else(|| default_tracker.as_ref().unwrap()));
+        let context = NodeBuilderContext::new(
+            flags.unwrap_or(NodeBuilderFlags::None),
```

Comment:
You might prefer the explicitness you have here, but `unwrap_or_default` could be nice if `NoneBuilderFlags` implemented `Default`.